packages/jaeger-ui-components/src/TraceTimelineViewer/SpanLinks.tsx

```diff
@@ -0,0 +1,121 @@
+
+import { SpanLinks } from '../types/links';
+
+interface SpanLinksProps {
+  links: SpanLinks;
+}
+
+const renderMenuItems = (links: SpanLinks, styles: ReturnType<typeof getStyles>, closeMenu: () => void) => {
+  return (
+    <>
+      {!!links.logLinks?.length ? (
+        <MenuGroup label="Logs">
+          {links.logLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label="Logs for this span"
+              onClick={(e) => {
+                if (link.onClick) {
+                  link.onClick(e);
+                }
+                closeMenu();
+              }}
+              url={link.href}
+              className={styles.menuItem}
+            />
+          ))}
+        </MenuGroup>
+      ) : null}
+      {!!links.metricLinks?.length ? (
+        <MenuGroup label="Metrics">
+          {links.metricLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label="Metrics for this span"
+              onClick={(e) => {
+                if (link.onClick) {
+                  link.onClick(e);
+                }
+                closeMenu();
+              }}
+              url={link.href}
+              className={styles.menuItem}
+            />
+          ))}
+        </MenuGroup>
+      ) : null}
+      {!!links.traceLinks?.length ? (
+        <MenuGroup label="Traces">
+          {links.traceLinks.map((link, i) => (
+            <MenuItem
+              key={i}
+              label={link.title ?? 'View linked span'}
+              onClick={(e) => {
+                if (link.onClick) {
+                  link.onClick(e);
+                }
+                closeMenu();
+              }}
+              url={link.href}
+              className={styles.menuItem}
+            />
+          ))}
+        </MenuGroup>
+      ) : null}
+    </>
+  );
+};
+
+export const SpanLinksMenu = ({ links }: SpanLinksProps) => {
+  const styles = useStyles2(getStyles);
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [menuPosition, setMenuPosition] = useState({ x: 0, y: 0 });
+
+  const closeMenu = () => setIsMenuOpen(false);
+
+  return (
+    <>
+      <button
+        onClick={(e) => {
+          setIsMenuOpen(true);
+          setMenuPosition({
+            x: e.pageX,
+            y: e.pageY,
+          });
+        }}
+        className={styles.button}
+      >
+        <Icon name="link" className={styles.button} />
+      </button>
+
+      {isMenuOpen ? (
+        <ContextMenu
+          onClose={() => setIsMenuOpen(false)}
+          renderMenuItems={() => renderMenuItems(links, styles, closeMenu)}
+          focusOnOpen={true}
+          x={menuPosition.x}
+          y={menuPosition.y}
+        />
+      ) : null}
+    </>
+  );
+};
+
+const getStyles = () => {
+  return {
+    button: css`
+      background: transparent;
+      border: none;
+      padding: 0;
+      margin: 0 3px 0 0;
+    `,
+    menuItem: css`
+      max-width: 60ch;
```

Comment:
Cool use of ch to measure the string length. I love the way it focuses on the 0 (zero) char for the current font to calc the width. No problem as I believe we only use the one font. Please lmk if we let the user choose the font for the menu in any way e.g. through a kind of global config somewhere.

Reply:
I don't think users have the ability to edit the font. If they do, the `ch` unit shouldn't vary too much. I can set a width in pixels if it's a concern.

Reply:
Ahh no, thanks though 👍